commit a1c2be2bfd08d639bb14f90540daff107848a26d
Author: Tor Didriksen <tor.didriksen@oracle.com>
Date:   Tue Jul 25 14:30:28 2023 +0200

    Bug #34278103 Compile MySQL with GCC 13 [noclose]
    
    Downgrade -Werror=suggest-attribute=format to -Wno-error
    
    Post-push fix: broken build for cmake . -DWITH_UNIT_TESTS=0
    
    Change-Id: I1edfff557741c353c56a3312a291df7e9b5297fa

diff --git a/storage/ndb/src/common/util/CMakeLists.txt b/storage/ndb/src/common/util/CMakeLists.txt
index b9b308f9d61..a68d64218aa 100644
--- a/storage/ndb/src/common/util/CMakeLists.txt
+++ b/storage/ndb/src/common/util/CMakeLists.txt
@@ -86,9 +86,11 @@ ENDFOREACH(tests)
 
 # This looks like a gcc bug, there *are* ATTRIBUTE_FORMAT annotations for
 # cstrbuf::appendf()
-IF(MY_COMPILER_IS_GNU AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 13)
-  TARGET_COMPILE_OPTIONS(cstrbuf-t PRIVATE
-    "-Wno-error=suggest-attribute=format")
+IF(WITH_UNIT_TESTS)
+  IF(MY_COMPILER_IS_GNU AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 13)
+    TARGET_COMPILE_OPTIONS(cstrbuf-t PRIVATE
+      "-Wno-error=suggest-attribute=format")
+  ENDIF()
 ENDIF()
 
 NDB_ADD_TEST(ndb_version-t version.cpp LIBS ndbgeneral)
